src/mlpack/tests/init_rules_test.cpp

```diff
@@ -197,6 +198,119 @@ BOOST_AUTO_TEST_CASE(GaussianInitTest)
   BOOST_REQUIRE_EQUAL(weights3d.n_slices, slices);
 }
 
+/**
+ * Simple test of the VarianceScalingNormalInit class.
+ */
+BOOST_AUTO_TEST_CASE(VarianceScalingNormalInitTest)
+{
+  const size_t rows = 7;
+  const size_t cols = 8;
+  const size_t slices = 2;
+
+  arma::mat weights;
+  arma::cube weights3d;
+
+  VarianceScalingNormalInit<> vsn;
+
+  vsn.Initialize(weights, rows, cols);
+  vsn.Initialize(weights3d, rows, cols, slices);
+
+
+  // Check that size of initialized weights is equal to the input weights size
+
+  BOOST_REQUIRE_EQUAL(weights.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights.n_cols, cols);
+
+  BOOST_REQUIRE_EQUAL(weights3d.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights3d.n_cols, cols);
+  BOOST_REQUIRE_EQUAL(weights3d.n_slices, slices);
+
+  // Manually create and initialize weight matrices with same dimensions and
+  // using the same initialization rule (Varaince Scaled Normal)
+
+  double variance = (1 / ( ( rows + cols ) / 2 ) );
+
+  arma::mat vsnWeights = arma::randn<arma::mat>(rows, cols);
+  arma::cube vsnWeights3d = arma::randn<arma::cube>(rows, cols, slices);
+
+  // Scale the weights from N(0,1) to N(0,variance) manually
+
+  vsnWeights *= sqrt(variance);
+  
+  vsnWeights3d *= sqrt(variance);
+  
+  // Check the values of initialized weights are close enough to manually 
+  // initialized weights for the matrix and cube
+
+  for (size_t i = 0; i < rows; i++)
+    for (size_t j = 0; j < cols; j++)
+      BOOST_REQUIRE_SMALL(weights.at(i, j) - vsnWeights.at(i, j), 1e-3);
+
+  for (size_t k = 0; k < slices; k++)
+    for (size_t i = 0; i < rows; i++)
+      for (size_t j = 0; j < cols; j++)
+        BOOST_REQUIRE_SMALL(weights3d.slice(k).at(i, j) - 
+        					vsnWeights3d.slice(k).at(i, j), 1e-3);
+
```

Comment:
I now realize the test I wrote (taken from OthogonalInit test without much thought) regarding how close the values should be to manually initializing the weights using the same approach doesn't make much sense since the range of numbers from the underlying distribution is much higher than 1e-3. 

@rcurtin Could you suggest me some other tests which should be useful in this case. Maybe I should add checks to see if the values are lying in the required range ? Any suggestions are most welcome.

Reply:
Yeah coming up with a meaningful test here is tricky, testing if they are in the correct range is one approach.

Reply:
Okay I will edit the tests to check the range which the initialized values lie in.